### PR TITLE
feat: [Guice 6] Migrate to use `jakarta.inject.*` for injection annotations

### PR DESCRIPTION
**Pull request by Nava2**
Part of #2863.

This replaces usages to enable migration to [Guice 7+](https://github.com/google/guice/wiki/Guice700). This is the first (only?) step to allow projects using Guice 6.0.0 or Guice 7.0.0 to use misk-inject.

By using `jakarta.inject` instead of `javax.inject`, injections work in both 6+ and 7+ versions. The new project, `:misk-inject-guice7-test` sets up a test project with the classpath forced to Guice 7+. This acts as a guard project for basic usages. As needed, we can add more content.

The exception is using `com.google.inject.Provider` as Guice 6 does not provide bindings that work with `jakarta.inject.Provider` in all cases, making this a frustrating requirement.

### CODE REVIEW

File: detekt.yaml [hunk 1]
```diff
@@ -21,6 +21,22 @@ complexity:
   LongParameterList:
     active: false
 
+style:
```
Comment (Nava2) [inline detekt.yaml:24]: Bans imports in future code

**Comment by Nava2** [comment]
cc @adrw / @r3mariano

**Review by r3mariano — APPROVED** [review]

**Comment by Nava2** [comment]
@r3mariano I'm unsure how to actually land this change. Could you advise on process or documentation?

**Comment by Nava2** [comment]
@jdm-square looks like there's some dependency cleanups and general lint things. Is there an autoformat somewhere.. was hoping the detekt formatting rules were in place but seem to not be. 

Also looks like I need to regenerate the API lockfiles, easy enough.

**Comment by jdm-square** [comment]
I don't see any linting issues that require formatting, only the missing api dump and dependency changes.

**Comment by r3mariano** [comment]
> @r3mariano I'm unsure how to actually land this change. Could you advise on process or documentation?

@Nava2 Nothing particular for the OSS repo. We're already aware of the change and can handle the change in our internal services (i.e. upgrade everyone to 6). Releases already happen automatically, we switched to a date-based versioning scheme that publishes for every change in main.

There is an incoming change to add more (non-Guice) modules, hopefully that won't cause a conflict.

**Comment by jdm-square** [comment]
```
CronTest > leaseDenied() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: 1
     but was: 0
        at java.base@17.0.7/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base@17.0.7/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
        at java.base@17.0.7/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base@17.0.7/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
        at app//misk.cron.CronTest.leaseDenied(CronTest.kt:78)
```
I'll run it again to see if it's an intermittent error.

**Comment by Nava2** [comment]
@jdm-square / @r3mariano could one of you enable the tests to run again? :) This should be bueno now!

**Comment by Nava2** [comment]
Thanks to whomever enabled the CI, I can't actually merge this :shipit:

**Pull request merged**